react/features/filmstrip/components/native/BaseIndicator.js

```diff
@@ -0,0 +1,51 @@
+// @flow
+
+import React, { Component } from 'react';
+import { View } from 'react-native';
+
+import { Icon } from '../../../base/font-icons';
+
+import styles from './styles';
+
+type Props = {
+
+    /**
+     * True if a highlighted background has to be applied.
+     */
+    highlight: boolean,
+
+    /**
+     * The name of the icon to be used as the indicator.
+     */
+    icon: string
+};
+
+/**
+ * Implements a base indicator to be reused across all native indicators on
+ * the filmstrip.
+ */
+export default class BaseIndicator extends Component<Props> {
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     */
+    render() {
+        const { highlight, icon } = this.props;
+
+        const iconElement = (<Icon
+            name = { icon }
+            style = { styles.indicator } />
+        );
+
+        if (highlight) {
```

Comment:
Can't we do this by using a different style to the view? Do you think it matters?

Reply:
We definitely need a container (View) to be able to apply a rounded border, but I managed to refactor this to be easier, thanks